rtl/templates/pinmux.sv.tpl

```diff
@@ -0,0 +1,107 @@
+// Copyright lowRISC contributors
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+// rtl/system/pinmux.sv is automatically generated using util/top_gen.py from rtl/templates/pinmux.sv.tpl
+// Please make any edits to the template file.
+
+module pinmux (
+  // Clock and reset
+  input logic clk_i,
+  input logic rst_ni,
+
+  // Blocks
+  % for label, width, name, instances in block_ios:
+  ${label} logic ${width}${name}[${instances}],
+  % endfor
+
+  // Pins
+  % for width, name in pin_ios:
+  inout logic ${width}${name},
```

Comment:
I agree I've essentially merged the padring and the pinmux. I'm not sure if splitting it up will make a cleaner design, but I'm not against it. Would it be ok to merge the PR as is and then potentially split the padring and the pinmux in a future PR?

Reply:
It's also mentioned in the pinmux refinement issue: https://github.com/lowRISC/sonata-system/issues/169

Reply:
It's about separation of concerns. The functionality of a pinmux is distinct from a pad instance/pad ring so it's better to de-couple them. It's particularly critical in this instance for two reasons:

1. Use of `inout` and `Z`. As I said in my earlier comment these need to be carefully handled in RTL. The pinmux is a large auto-generated module so there's lot to look at and think about so don't want to add ensuring correct usage of tristate as part of this. By keeping the pinmux strictly digital logic and isolating the tristate stuff into a per pad module there's a far smaller bit of RTL to review for correct tristate usage and lower maintenance burden going forward (we only have to think about tristate when we touch the pad instance). It would have reduced the effort required on my part to review this for instance.

2. Ease of switching pad implementation. Different targets (different FPGAs, ASIC etc) may want different pad instances. The generic `assign pad = pad_oe ? pad_o : 'z`  does do the job for FPGA most of the time but there may be reasons you want a technology specific pad instance and it would be essential were this ever to be used against an ASIC target.

Happy for this to get merged without this getting changed but I don't want it dropped as a 'nice-to-have' task that doesn't get done.